android/src/main/java/com/pinmi/react/printer/adapter/BLEPrinterAdapter.java

```diff
@@ -188,9 +176,53 @@ public void run() {
     @Override
     public void printByteData(byte[] rawBase64Data, Callback errorCallback) {
     }
-
+    
+    @Override
+    public void connectAndSend(String host, Integer port, String rawBase64Data, Callback successCallback, Callback errorCallback) {
+    }
 
     @Override
-    public void connectAndSend(String host, Integer port, String data, Callback successCallback, Callback errorCallback) {
+    public void connectAndSend(PrinterDeviceId printerDeviceId, String data, Callback successCallback, Callback errorCallback) {
+      BluetoothAdapter bluetoothAdapter = getBTAdapter();
+      if(bluetoothAdapter == null) {
+        errorCallback.invoke("No bluetooth adapter available");
+        return;
+      }
+      if (!bluetoothAdapter.isEnabled()) {
+        errorCallback.invoke("bluetooth is not enabled");
+        return;
+      }
+      BLEPrinterDeviceId blePrinterDeviceId = (BLEPrinterDeviceId)printerDeviceId;
+      if(this.mBluetoothDevice != null){
+        if(this.mBluetoothDevice.getAddress().equals(blePrinterDeviceId.getInnerMacAddress()) && this.mBluetoothSocket != null){
+          Log.v(LOG_TAG, "do not need to reconnect");
+          printRawData(data, errorCallback);
+          successCallback.invoke(new BLEPrinterDevice(this.mBluetoothDevice).toRNWritableMap());
+          return;
+        }else{
+          closeConnectionIfExists();
+        }
+      }
+      Set<BluetoothDevice> pairedDevices = getBTAdapter().getBondedDevices();
```

Comment:
what are all these lines of code doing?

I see we are already printing in Line  199

Reply:
These list all the available bluetooth devices and loop over them and find whichever matches

Reply:
The line 199 is inside if block. If already matching device is present